### PR TITLE
#69 refactor: created person resource and swapped it with person_id in ca…

### PR DESCRIPTION
**Pull request by fkula5**
…st resource

### CODE REVIEW

File: app/Http/Resources/CastResource.php [hunk 1]
```diff
@@ -17,7 +18,7 @@ public function toArray($request)
         return [
             'id' => $this->id,
             'medium_id' => $this->medium_id,
-            'person_id' => $this->person_id,
+            'person_data' => new PersonResource(Person::findOrFail($this->id)),
```
Comment (marioooo0o) [inline app/Http/Resources/CastResource.php]: Wrong! You are looking for a person with a role id, not a person id!!!
Additionally in the cast model you have the person() function which allows you to get exactly this person without giving any keys. (there is no possibility of a mistake you inadvertently made)
Finally it should look like this:
'person' => new PersonResource($this->person),

File: app/Http/Resources/CastResource.php [hunk 2]
```diff
@@ -17,7 +18,7 @@ public function toArray($request)
...
             'gender_id' => $this->gender_id,
```
Comment (marioooo0o) [inline app/Http/Resources/CastResource.php]: Do the same with gender that you use resource

File: app/Http/Resources/CastResource.php [hunk 3]
```diff
@@ -17,7 +18,7 @@ public function toArray($request)
...
             'character' => $this->character,
             'created_at' => $this->created_at->format('d-m-Y'),
```
Comment (marioooo0o) [inline app/Http/Resources/CastResource.php:23]: format ok but look for how to add the month in words

**Review by marioooo0o — CHANGES_REQUESTED** [review]
Cardinal mistake you have to watch out for ;)

File: app/Http/Resources/CastResource.php (hunk 3 above)
Comment (fkula5) [reply app/Http/Resources/CastResource.php:23]: We decided to use number and then on the frontend we gonna transfer it to month name

Comment (marioooo0o) [reply app/Http/Resources/CastResource.php:23]: Ok then no problem :D

File: app/Http/Resources/CastResource.php [hunk 4]
```diff
@@ -18,8 +18,8 @@ public function toArray($request)
         return [
             'id' => $this->id,
             'medium_id' => $this->medium_id,
-            'person_data' => new PersonResource(Person::findOrFail($this->id)),
-            'gender_id' => $this->gender_id,
+            'person_data' => new PersonResource($this->person),
+            'gender_id' => $this->gender,
```
Comment (marioooo0o) [inline app/Http/Resources/CastResource.php]: This isn't gender_id but gender model

Comment (marioooo0o) [reply app/Http/Resources/CastResource.php]: Additionally you were supposed to return it in resource

File: app/Http/Resources/CastResource.php [hunk 5]
```diff
@@ -18,8 +18,8 @@ public function toArray($request)
         return [
             'id' => $this->id,
             'medium_id' => $this->medium_id,
-            'person_data' => new PersonResource(Person::findOrFail($this->id)),
-            'gender_id' => $this->gender_id,
+            'person_data' => new PersonResource($this->person),
```
Comment (marioooo0o) [inline app/Http/Resources/CastResource.php]: please change variable to 'person'

**Review by marioooo0o — CHANGES_REQUESTED** [review]
one more correction

File: app/Http/Resources/CastResource.php [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 namespace App\Http\Resources;
 
+use App\Models\Person;
```
Comment (marioooo0o) [inline app/Http/Resources/CastResource.php]: Why do you need Person model?

**Review by marioooo0o — CHANGES_REQUESTED** [review]
Please delete this import and I will merge this PR

**Review by marioooo0o — APPROVED** [review]
Sounds good. I approve 👍🏻

**Pull request merged**